### DISCUSSION TITLE
iina gives error with youtube url invoked through hammerspoon but mpv works

### DISCUSSION THREAD

**Discussion by jdsutherland** [iina/iina#5252] [Q&A] [answered]
I wanted to bind a key to open a youtube url from clipboard in iina but it results in `Error: Cannot open file or stream!` but the identical arguments using mpv work:

To reproduce from hammerspoon console:
`hs.execute('mpv "https://www.youtube.com/watch?v=1B1dnSf6rPk"')` (works)

`hs.execute('/opt/homebrew/bin/iina "https://www.youtube.com/watch?v=1B1dnSf6rPk" 2>&1')` (Error: Cannot open file or stream!)
`2024-11-05 13:53:08.655 IINA[86796:610312] +[IMKClient subclass]: chose IMKClient_Modern
2024-11-05 13:53:08.655 IINA[86796:610312] +[IMKInputSession subclass]: chose IMKInputSession_Modern
IINA 1.3.5 Build 141
	true	exit	0`

Is it because iina is a gui and has some macos limitations? I see that `open a -IINA` doesn't seem to support urls so it seems I have to use mpv to achieve this.

**Comment by low-batt** [comment, marked as answer]
Try adding the `--no-stdin` option. From `--help`:
```text
--stdin, --no-stdin:
        You may also pipe to stdin directly. Sometimes iina-cli can detect whether
        stdin has file, but sometimes not. Therefore it's recommended to always
        supply --stdin when piping to iina, and --no-stdin when you are not intend
        to use stdin.
```

Certain ways of invoking can confuse the automatic detection causing IINA decide it needs to read from stdin. If the video file is not being piped into IINA then that will trigger the `Cannot open file or stream!` error.

**Reply by jdsutherland** [reply]
That works. Thanks.

**Reply by low-batt** [reply]
Glad to hear that worked. I only knew about this option because recently a user on IINA's Telegram channel encountered a similar problem getting IINA to work from a script.
